index.js

```diff
@@ -106,6 +118,11 @@
     return f(this.value);
   };
 
+  //  Just#concat :: m a -> m a
+  Just.prototype.concat = function(maybe) {
+    return maybe instanceof Just ? Just(this.value.concat(maybe.value)) : this;
+  };
+
```

Comment:
> There are plenty of ways to shoot yourself in the foot with js. Is there a reason `Semigroup`/`Monoid` should be held to some stronger requirements?

No, but I was hoping that `Maybe` could be.  My reason for adding `Maybe` to a system that otherwise doesn't have one is to prevent having to do all sorts of error-checking across the system, or, in failing to do that, allowing for the possibility of exceptions arising.  So I want my `Maybe` to be as bullet-proof as possible.

But I finally went back and looked at the `filter` implementation involved, and perhaps I've been confused by the discussion.  It looks as though `filter` has nothing to do with `Semigroups` or `Monoids`.  There seems to be what might be seen to be a pun on `empty` here.  There seems to be nothing in the definition of `filter`, explicit or implied, that requires `empty` be a lawful Monoid identity.  `filter` only requires that such a method exists (and presumably requires that it returns a value of the same `Maybe`.)  Moreover all my concerns above had to do with the definition of `concat`, and the fact that it could easily throw unexpected exceptions.  But that's entirely irrelevant, since the definition of `filter` being discussed has nothing at all to do with the `Maybe` being a `Semigroup` or a `Monoid`.

So my take again, would be not to add the `Semigroup` or `Monoid` without good reason, but since they wouldn't actually get invoked during `filter`, I'm less bothered by them.